src/pointfree/toPromise.js

```diff
@@ -0,0 +1,14 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis */
+
+const isFunction = require('../core/isFunction')
+
+function toPromise(m) {
+  if(!(m && isFunction(m.toPromise))) {
```

Comment:
As this is only for the `Async` type, this should live in `src/Async/toPromise`. Then as it is in the same folder `isSameType` should be used like so:

```js
const Async = require('./index.js')
const isSameType = require('../core/isSameType.js')

function toPromise(m) {
  if(!isSameType(Async, m)) {
    throw new TypeError('toPromise: Async required')
  }
}
```
While you the type usually come in as `require('./')`, due to the work started by you and @bennypowers around modules, we should start being very explicit about file names and extensions going forward. You do not have to do it for this PR mind you, just something to keep in mind as we move forward.

Reply:
Haha this is funny, thats where i had it originally using isSameType and then changed it because i thought you'd want it like the other ones like run or map :)

Reply:
if it was a generic pointfree function, then I would say this is the proper inplementation. You totally nailed it. But it is specific to one type then the pattern should be using isSameType like [here](https://github.com/evilsoft/crocks/blob/master/src/Async/race.js#L10)